### PR TITLE
StatsVC layout improvements for dynamic type 

### PR DESCRIPTION
**Pull request by MelanijaGrunte**
Includes DIG-570 and DIG-571

### CODE REVIEW

File: apturicovid/viewController/StatsVC.swift [hunk 1]
```diff
@@ -124,6 +130,20 @@ extension StatsVC : UICollectionViewDataSource {
...
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        if let header = self.collectionView(collectionView, viewForSupplementaryElementOfKind: UICollectionView.elementKindSectionHeader, at: IndexPath(row: 0, section: section)) as? StatsHeaderView {
+            
+            let titleLabelSize = header.titleLabel.sizeThatFits(CGSize(width: collectionView.frame.width, height: .greatestFiniteMagnitude))
+            let descriptionLabelSize = header.descriptionLabel.sizeThatFits(CGSize(width: collectionView.frame.width, height: .greatestFiniteMagnitude))
+            
+            let height = titleLabelSize.height + header.titleLabelTopConstraint.constant + descriptionLabelSize.height + header.descriptionLabelTopConstraint.constant
+                        
+            return CGSize(width: collectionView.frame.width, height: height)
+        }
```
Comment (ULazdins) [inline apturicovid/viewController/StatsVC.swift:157]: Add else here please

File: apturicovid/viewController/StatsVC.swift [hunk 2]
```diff
@@ -124,6 +130,20 @@ extension StatsVC : UICollectionViewDataSource {
         stats?.totalItemCount ?? 0
     }
     
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        if let header = self.collectionView(collectionView, viewForSupplementaryElementOfKind: UICollectionView.elementKindSectionHeader, at: IndexPath(row: 0, section: section)) as? StatsHeaderView {
+            
+            let titleLabelSize = header.titleLabel.sizeThatFits(CGSize(width: collectionView.frame.width, height: .greatestFiniteMagnitude))
+            let descriptionLabelSize = header.descriptionLabel.sizeThatFits(CGSize(width: collectionView.frame.width, height: .greatestFiniteMagnitude))
+            
+            let height = titleLabelSize.height + header.titleLabelTopConstraint.constant + descriptionLabelSize.height + header.descriptionLabelTopConstraint.constant
```
Comment (ULazdins) [inline apturicovid/viewController/StatsVC.swift]: Can't we use autolayout here?

File: apturicovid/viewController/StatsVC.swift [hunk 3]
```diff
@@ -124,6 +130,17 @@ extension StatsVC : UICollectionViewDataSource {
         stats?.totalItemCount ?? 0
     }
     
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        if let headerView = self.collectionView(collectionView, viewForSupplementaryElementOfKind: UICollectionView.elementKindSectionHeader, at: IndexPath(row: 0, section: section)) as? StatsHeaderView {
+            
+            return headerView.systemLayoutSizeFitting(CGSize(width: collectionView.frame.width, height: UIView.layoutFittingCompressedSize.height),
+                                                      withHorizontalFittingPriority: .required,
+                                                      verticalFittingPriority: .fittingSizeLevel)
```
Comment (MelanijaGrunte) [inline apturicovid/viewController/StatsVC.swift:154]: @ULazdins UICollectionView's auto-layout can be achieved through code only. I've updated the method.

**Review by ULazdins — APPROVED** [review]

**Comment by ArtSp** [comment]
Unable to simultaneously satisfy constraints in collection header view.

**Comment by ArtSp** [comment]
Collection view footer is not scaling.

**Review by ArtSp — CHANGES_REQUESTED** [review]
- Unable to simultaneously satisfy constraints in collection header view.

- Collection view footer is not scaling

- Collection view doesn't reload when text scale changes. 

- Single row layout must be implemented only for accessibility scales. Display zoom must not be used to determine row count , Plus size phones have full lot of space even with zoomed screen. 

- Cell title label has no top/bottom constraints. With max accessibility scales text height ~= superview height. Can use ~ >=8pt constraints.

- It's better to embed cell value description and value labels into UIStackViews. For larger accessibility fonts you can use vertical axis, and use horizontal for others. Otherwise on SE screen every label has it's own font size, as they all try to fit width. 

- Use `superView.backgroundColor = welcomeHeaderView.fillColor` when collection view content height > than welcomeHeaderView height. Previously this was used only for SE 1gen screens. 

- Align cell value description labels leading anchor to cell description label's leading anchor.

- Adjust cell size to font scale.

File: apturicovid/viewController/StatsVC.swift [hunk 4]
```diff
@@ -46,6 +54,8 @@ class StatsVC: BaseViewController {
         statsCollectionView.refreshControl = refreshControl
         refreshControl.layer.zPosition = -1
         
+        NotificationCenter.default.addObserver(self, selector: #selector(preferredContentSizeChanged(_:)), name: UIContentSizeCategory.didChangeNotification, object: nil)
```
Comment (mzibara) [inline apturicovid/viewController/StatsVC.swift]: Couldn't we do this with rxcocoa as well?

File: apturicovid/viewController/StatsVC.swift [hunk 5]
```diff
@@ -50,7 +50,12 @@ class StatsVC: BaseViewController {
         statsCollectionView.refreshControl = refreshControl
         refreshControl.layer.zPosition = -1
         
-        NotificationCenter.default.addObserver(self, selector: #selector(preferredContentSizeChanged(_:)), name: UIContentSizeCategory.didChangeNotification, object: nil)
+        NotificationCenter.default.rx
+            .notification(UIContentSizeCategory.didChangeNotification)
+            .subscribe(onNext: { [weak self] (_) in
```
Comment (ArtSp) [inline apturicovid/viewController/StatsVC.swift:56]: Add .observeOn(MainScheduler.instance) so there will be no need to call DispatchQueue.main in preferredContentSizeChanged()

**Review by ArtSp — APPROVED** [review]

**Pull request merged**